npono21/mini-shopify
# Feature/53 shop search functionality

## PR type


## Description

This MR is motivated by the need to be able to search Shops by Name and by Tag.

This MR introduces the following changes:
* Implements a SearchController to handle search requests
* Adds Tags to Shops, and updates constructor and calls where needed
* Adds Tag checklist to Shop creation form on Merchant home page
* Enables search by Shop Tags and Names, and returns a tiled view of matching Shops
* Updates relevant unit tests
* Updates existing HTML documents to enable the search bar on those pages
* Adds this use case to `README`

__Notes__:
* Issuing this PR, but acknowledging I will have many conflicts after #62 is merged.

## Related Issues/Tickets

closes #53 #60
## QA Instruction, Screenshots etc...

To test search functionality:
1. Register as a Merchant.
2. Create one or more Shops, selecting Tags during Shop creation.
3. Enter your search terms in the Search bar at the top of the Merchant home page.
4. View results and verify.

All Shops with substring matches of any of your search terms (whitespace-separated) with respect to the Shop Name or Shop Tags should be returned and displayed on the Search Results page.


## Added/updated tests?

- [x] Yes: Updated some unit tests to include the new Tags field on Shops.
- [ ] No: _replace this line with details on why tests are not included in this PR_

## PR Checklist

- [x] I have rebased this branch against main
- [x] My commit messages are descriptive and useful
